web/src/compose_validate.js

```diff
@@ -59,7 +59,16 @@ export function needs_subscribe_warning(user_id, stream_id) {
     return true;
 }
 
-function get_stream_id() {
+function get_stream_id_for_textarea($textarea) {
+    const is_in_editing_area = $textarea.closest(".message_row").length > 0;
+
+    if (is_in_editing_area) {
+        return Number.parseInt(
+            $textarea.closest(".recipient_row").find(".message_header").attr("data-stream-id"),
+            10,
+        );
+    }
+
     const stream_name = compose_state.stream_name();
 
     if (!stream_name) {
```

Comment:
This stream of logic to get the stream ID is pretty messy, round-tripping through the stream name ... and we do this in like 5 places in the compose code path right. I'm pretty sure with how the compose box stream picker component works today, the thing we're actually storing at a raw level of is the stream ID... we just haven't gotten around to rewriting the `compose_state` interface to return that.

I opened https://github.com/zulip/zulip/issues/25742 to track this.